ADA/databricks_rstudio_sql_warehouse.qmd

```diff
@@ -1,105 +1,226 @@
+The Databricks SQL warehouse ID is the warehouse containing data that you would like to use. To get the warehouse ID, follow these steps:
 
-    i)  Login to [Databricks](https://adb-6882499576863257.17.azuredatabricks.net/?o=6882499576863257)
+-   click 'SQL Warehouses' under the 'SQL' section of the left hand menu on Databricks
+-   click on the warehouse name that you'd like to get the ID for
+-   the warehouse ID is in brackets next to the warehouse name on the 'Overview' tab
 
-    ii) Click on the 'SQL Warehouses' tab in the sidebar. ![](../images/databricks-SQL-warehouses.png)
+------------------------------------------------------------------------
 
-    iii) Click on the name of the warehouse you want to connect to, and click the 'connection details' tab.
+#### Databricks token
 
-    iv) Make a note of the 'Server hostname', 'Port', and 'HTTP Path'.
+------------------------------------------------------------------------
 
-3.  Get a personal access token from Databricks for authentication.
+The Databricks token is a personal access token. 
 
-    i)  In Databricks, click on your email address in the top right corner, then click 'User settings'.
+A personal access token is is a security measure that acts as an identifier to let Databricks know who is accessing information from the SQL warehouse. Access tokens are usually set for a limited amount of time, so they will need renewing periodically.
 
-    ii) Go to the 'Developer' tab in the side bar. Next to 'Access tokens', click the 'Manage' button. ![](../images/databricks-access-tokens.png)
+-   In Databricks, click on your email address in the top right corner, then click 'User settings'
 
-    iii) Click the 'Generate new token' button.
+-   Go to the 'Developer' tab in the side bar. Next to 'Access tokens', click the 'Manage' button
 
-    iv) Name the token, then click 'Generate'. *Note that access tokens will only last as long as the value for the 'Lifetime (days)' field. After this period the token will expire, and you will need to create a new one to re-authenticate.*
+::: {align="center"}
+![](../images/databricks-access-tokens.png)
+:::
 
-    v)  Make a note of the 'Databricks access token' it has given you. **It is important to copy this somewhere as you will not be able to see it through Databricks again.**
+-   Click the 'Generate new token' button
 
-4.  Setup ODBC connection from your laptop. We now have all the information we need to setup a connection between our laptop and DataBricks.
+-   Name the token, then click 'Generate'
 
-    i)  In the start menu, search 'ODBC' and open 'ODBC Data Sources (64-bit)'.
+::: callout-note
+Note that access tokens will only last as long as the value for the 'Lifetime (days)' field. After this period the token will expire, and you will need to create a new one to re-authenticate.
```

Comment:
We could mention the option of leaving 'Lifetime' blank and having the token never expire. Maybe we shouldn't be suggesting it though since best practice is to have tokens expire once in a while. I'll defer to others.

Reply:
I don't think security would appreciate this. They were designed to expire to prevent old forgotten / unused tokens kicking around.